clients/drcachesim/tools/view.cpp

```diff
@@ -573,14 +572,16 @@ view_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
          * disassemble DR_ISA_REGDEPS instructions.
          */
         dr_isa_mode_t old_isa_mode = dr_get_isa_mode(dcontext_.dcontext);
-        if (TESTANY(OFFLINE_FILE_TYPE_ARCH_REGDEPS, filetype_)) {
+        bool is_regdeps_trace =
+            filetype_ != -1 && TESTANY(OFFLINE_FILE_TYPE_ARCH_REGDEPS, filetype_);
```

Comment:
Change the init value to 0?  Not ideal to have to do this extra check.

Reply:
But 0 is `OFFLINE_FILE_TYPE_DEFAULT`.
I'll check if it's safe to do so, I haven't considered the side-effects of that.

Reply:
Not relevant to this PR anymore.